src/com/owncloud/android/ui/fragment/EditShareFragment.java

```diff
@@ -186,19 +186,22 @@ private void refreshUiFromState(View editShareView) {
             boolean canEdit = (sharePermissions & anyUpdatePermission) > 0;
             compound.setChecked(canEdit);
 
-            if (mFile.isFolder() && (!isFederated || isNotReshareableFederatedSupported)) {
-                /// TODO delete extra && !isFederated when iOS is ready
+            boolean areEditOptionsAvailable = !isFederated || isNotReshareableFederatedSupported;
+
+            if (mFile.isFolder() && areEditOptionsAvailable) {
+                /// TODO change areEditOptionsAvailable in order to delete !isFederated
+                // from checking when iOS is ready
                 compound = (CompoundButton) editShareView.findViewById(R.id.canEditCreateCheckBox);
                 compound.setChecked((sharePermissions & OCShare.CREATE_PERMISSION_FLAG) > 0);
-                compound.setVisibility((canEdit && !isFederated) ? View.VISIBLE : View.GONE);
+                compound.setVisibility((canEdit && areEditOptionsAvailable) ? View.VISIBLE : View.GONE);
```

Comment:
In this point ```areEditOptionsAvailable``` is always ```true```, the variable can be deleted from the condition.

Same for lines 200 and 204.